src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -493,17 +493,33 @@ internal static string TupleMemberName(int position)
             return "Item" + position;
         }
 
+        // This array needs to be sorted for binary searching
+        private static readonly string[] reservedMemberNames = new[]
+        {
```

Comment:
After our discussion, I took a look at this idea (which we had discussed with Aleksey and Neal as well). It is tricky to check against underlying type because of the order of checks. We currently check the names, then build the tuple symbol (including the underlying type).
I'll leave as blacklist for now (with a `switch` as you suggested). We can revisit.